src/Loggers/OwnJSONPatternFormatter.cpp

```diff
@@ -62,6 +69,26 @@ void OwnJSONPatternFormatter::formatExtended(const DB::ExtendedLogMessage & msg_
     const Poco::Message & msg = msg_ext.base;
     DB::writeChar('{', wb);
 
+    if (!date_time_utc.empty())
+    {
+        writeJSONString(date_time_utc, wb, settings);
+        DB::writeChar(':', wb);
+
+        DB::writeChar('\"', wb);
+        static const DateLUTImpl & utc_time_zone = DateLUT::instance("UTC");
+        /// Change delimiters in date for compatibility with old logs.
+        writeDateTimeTextISO(msg_ext.time_seconds, 0, wb, utc_time_zone);
+        DB::writeChar('.', wb);
+        DB::writeChar('0' + ((msg_ext.time_microseconds / 100000) % 10), wb);
+        DB::writeChar('0' + ((msg_ext.time_microseconds / 10000) % 10), wb);
+        DB::writeChar('0' + ((msg_ext.time_microseconds / 1000) % 10), wb);
+        DB::writeChar('0' + ((msg_ext.time_microseconds / 100) % 10), wb);
+        DB::writeChar('0' + ((msg_ext.time_microseconds / 10) % 10), wb);
+        DB::writeChar('0' + ((msg_ext.time_microseconds / 1) % 10), wb);
```

Comment:
Unfortunately this does not produce RFC 3339 compliant datetime strings. The "parts of second" should appear _before_ the `Z`. (see https://it-tools.tech/date-converter)

eg. it should be `2024-10-31T15:20:50.281688Z` (not `2024-10-31T15:20:50Z.281688`)


If this is a limitation of `writeDateTimeTextISO` then I would suggest dropping the sub-second precision from this PR, and just logging it the second-level for now (eg. `2024-10-31T15:20:50Z`)

Reply:
maybe `writeDateTimeText` can be used?

Reply:
Thank you for pointing this out.
Fixed the issue.
Now: "2024-11-04T19:01:54Z"
Technically "writeDateTimeText" could be used as well but this is 8601 (RFC3339).